From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 22 Aug 2013 15:12:37 +0200
Subject: [PATCH] v4l2: filter: fix shutdown handling

Is this correct? It seems to remove some race conditions.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2filter.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/sys/v4l2/gstv4l2filter.c b/sys/v4l2/gstv4l2filter.c
index 7d94c53..913df4a 100644
--- a/sys/v4l2/gstv4l2filter.c
+++ b/sys/v4l2/gstv4l2filter.c
@@ -705,10 +705,7 @@ gst_v4l2filter_activate_mode (GstPad * pad, GstObject * parent, GstPadMode mode,
         g_mutex_unlock (&v4l2filter->queue_lock);
         /* FIXME: is this correct? */
         gst_v4l2_object_unlock (v4l2filter->src_v4l2object);
-        if (v4l2filter->src_v4l2object->pool) {
-          gst_buffer_pool_set_active (v4l2filter->src_v4l2object->pool, FALSE);
-          result = gst_pad_stop_task (pad);
-        }
+        result = gst_pad_stop_task (pad);
         gst_v4l2_object_unlock_stop (v4l2filter->src_v4l2object);
       }
       GST_V4L2FILTER_PREROLL_UNLOCK (v4l2filter);
@@ -781,7 +778,6 @@ gst_v4l2filter_sink_chain (GstPad * pad, GstObject * parent, GstBuffer * buffer)
   GST_V4L2FILTER_PREROLL_UNLOCK (v4l2filter);
 
   if (!GST_PAD_TASK (v4l2filter->srcpad)) {
-    gst_buffer_unref (buffer);
     return GST_FLOW_FLUSHING;
   }
 
